packages/@webex/plugin-meetings/src/multistream/sendSlotManager.ts

```diff
@@ -103,17 +113,39 @@ export default class SendSlotManager {
       throw new Error(`Slot for ${mediaType} does not exist`);
     }
 
+    const currentStateOverride = this.getSourceStateOverride(mediaType);
+    if (currentStateOverride === state) {
+      return;
+    }
+
     if (state) {
       slot.setSourceStateOverride(state);
+      this.sourceStateOverrides.set(mediaType, state);
     } else {
       slot.clearSourceStateOverride();
+      this.sourceStateOverrides.delete(mediaType);
     }
 
     this.LoggerProxy.logger.info(
       `SendSlotsManager->setSourceStateOverride#set source state override for ${mediaType} to ${state}`
     );
   }
 
+  /**
+   * Gets the source state override for the given media type.
+   * @param {MediaType} mediaType - The type of media to get the source state override for.
+   * @returns {StreamState | null} - The current source state override or null if not set.
+   */
+  public getSourceStateOverride(mediaType: MediaType): StreamState | null {
```

Comment:
This doesn't need to be public right? No one else needs to use this other than `SendSlotManager`?